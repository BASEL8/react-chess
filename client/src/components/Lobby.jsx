import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import ClearIcon from '@material-ui/icons/Clear';

const useStyles = makeStyles({
  wrapper: {
    diplay: 'flex',
    justifyContent: 'flex-start',
    flexWrap: 'nowrap'
  },
  table: {
    maxWidth: '60vw',
  },
  buttonContainer: {
    maxWidth: '25vw',
    display: 'flex',
    flexDirection: 'column',
    alignContent: 'center'
  }
});

function createData(id, player, fat, carbs, protein) {
  return { id, player, fat, carbs, protein };
}

const rows = [
  createData(1, 'Frozen yoghurt', 6.0, 24),
  createData(2, 'Ice cream sandwich', 9.0, 37),
  createData(3, 'Eclair', 16.0, 24),
  createData(4, 'Cupcake', 3.7, 67),
  createData(5, 'Gingerbread', 16.0, 49),
];

export default function Lobby () {
  const classes = useStyles();

  return (
    <div className={classes.wrapper}>
      <TableContainer component={Paper} className={classes.table}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>No.</TableCell>
              <TableCell>Player</TableCell>
              <TableCell>Play</TableCell>
              <TableCell>Delete</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map(row => (
              <TableRow key={row.id}>
                <TableCell component="th" scope="row">
                  {row.id}
                </TableCell>
                <TableCell>{row.player}</TableCell>
                <TableCell><Button variant="outlined">Play</Button></TableCell>
                <TableCell><Button><ClearIcon /></Button></TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
      <div className={classes.buttonContainer}>
        <Button variant="outlined">Create a new match</Button>
        <Button variant="outlined">Play with a friend</Button>
        <Button variant="outlined">Play against computer</Button>
      </div>
    </div>
  );
}
