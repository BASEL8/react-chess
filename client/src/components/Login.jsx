
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { Redirect, Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import { Grid, CssBaseline, Paper, Typography, Avatar, FormControlLabel, Checkbox, Box, TextField, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import {ReactComponent as Crown} from '../resources/crown.svg';

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright Â© '}
      <Link color="inherit" href="https://material-ui.com/">
        Chess Online
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles(theme => ({
  root: {
    height: '100vh',
  },
  image: {
    //backgroundImage: 'url(https://source.unsplash.com/random)',
    backgroundRepeat: 'no-repeat',
    backgroundColor:
      theme.palette.type === 'dark' ? theme.palette.grey[900] : theme.palette.grey[50],
    backgroundSize: 'cover',
    backgroundPosition: 'center',
  },
  paper: {
    margin: theme.spacing(8, 4),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.primary.main,
  },
  logo: {
    height: '5vh',
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

const Login = () => {
  const classes = useStyles();

  const [username, setUsername] = useState([]); 
  const [errorMessage, setErrorMessage] = useState('');
  const [auth, setAuth] = useState(false);
  const [isChecked, setIsChecked] = useState(false);

  // Handling username input
  const onChangeUsername = (e) => {
    setUsername(e.target.value);
  };

  // Handlling form submit
  const handleAddUser = (e) => {
    e.preventDefault();
    //localStorage.setItem('userId', username )
    axios.post('http://localhost:8000/api/seeks/', { userId: username })
      .then((res) => {      
        //setUsername('');
        setAuth(true);
      })
      .catch((error) => {
        if (error.response.status === 400) {
          setErrorMessage('Please fill out your name');
          setTimeout(() => { setErrorMessage(''); }, 2000); // message will disappear after 2000sec.
        }
      });
  };

  const toggleChange = () => {
    setIsChecked(!isChecked);
  }

  if (auth) {
    return (
      <Redirect to="/lobby" />
    );
  }

  if (!isChecked) {
    return (
      <>
      <Helmet><title>Chess online</title></Helmet>
      <Grid container component="main" className={classes.root}>
        <CssBaseline />
        <Grid item xs={false} sm={4} md={7} className={classes.image} />
        <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
          <div className={classes.paper}>
            <Crown className={classes.logo} />
  {/*           <Avatar className={classes.avatar}>
              <Crown />
            </Avatar> */}
            <Typography component="h1" variant="h5">
              Chess
            </Typography>
            <FormControlLabel
              control={<Checkbox color="primary" />}
              label="Add player"
              checked={isChecked}
          onChange={toggleChange}
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              data-testid="button"
              href="/lobby"
            >
              Enter
            </Button>
            <Box mt={5}>
              <Copyright />
            </Box>
          </div>
        </Grid>
      </Grid>
      </>
    );
  }else{
  return (
    <>
    <Helmet><title>Chess online</title></Helmet>
    <Grid container component="main" className={classes.root}>
      <CssBaseline />
      <Grid item xs={false} sm={4} md={7} className={classes.image} />
      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
          <Crown className={classes.logo} />
{/*           <Avatar className={classes.avatar}>
            <Crown />
          </Avatar> */}
          <Typography component="h1" variant="h5">
            Chess
          </Typography>
       <FormControlLabel
              control={<Checkbox color="primary" />}
              label="Add player"
              checked={isChecked}
          onChange={toggleChange}
            />
          <form className={classes.form} onSubmit={handleAddUser}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="userId"
              label="Player name"
              name="userId"
              helperText="Enter your name to play chess online"
              autoComplete="name"
              onChange={onChangeUsername}
              autoFocus
            />
            <p style={{ color: 'red' }}>{errorMessage}</p>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              data-testid="button"
            >
              Enter
            </Button>
            <Box mt={5}>
              <Copyright />
            </Box>
          </form>
        </div>
      </Grid>
    </Grid>
    </>
  );
        }
}
export default Login;
