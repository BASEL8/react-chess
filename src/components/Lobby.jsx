/* eslint-disable no-undef */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Helmet from 'react-helmet';

const Lobby = () => {
  const [seeks, setSeeks] = useState([]);

  useEffect(() => {
    axios.get(`http://localhost:8000/api/seeks/${localStorage.getItem('userId')}`)
      .then((response) => {
        // console.log(response.data);
        setSeeks(response.data);
      });
  }, []);

  if (!seeks) {
    return <p>Loading seeks...</p>;
  }

  return (
    <>
      <Helmet><title>Lobby</title></Helmet>
      <table>
        <thead>
          <tr>
            <th>Player One</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {seeks.map((seek) => (
            <>
              <tr key={seek.id}>
                <td>{seek.playerOne}</td>
                <td>
                  <button type="submit">Play</button>
                  {' '}
                </td>
              </tr>
            </>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Lobby;
