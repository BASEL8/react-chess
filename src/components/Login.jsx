
import React, { useState } from 'react';
import axios from 'axios';
import { Redirect } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import {
  TextField, Button,
} from '@material-ui/core';

const Login = () => {
  const [username, setUsername] = useState([]);
  const [errorMessage, setErrorMessage] = useState('');
  const [auth, setAuth] = useState(false);

  // Handling username input
  const handleUsername = (e) => {
    setUsername(e.target.value);
  };

  // Handlling form submit
  const handleAddUser = (e) => {
    e.preventDefault();
    axios.post('http://localhost:8000/api/seeks/', { userId: username })
      .then(() => {
        setUsername('');
        setAuth(true);
      })
      .catch((error) => {
        if (error.response.status === 400) {
          setErrorMessage('You must fill in username');
          setTimeout(() => { setErrorMessage(''); }, 2000); // message will disappear after 2000sec.
        }
      });
  };

  if (auth) {
    return (
      <Redirect to="/lobby" />
    );
  }
  return (
    <div>
      <Helmet><title>Login</title></Helmet>
      <div className="login-form">
        <form onSubmit={handleAddUser}>
          <p style={{ color: 'red' }}>{errorMessage}</p>
          <TextField id="standard-basic" label="Username" name="username" placeholder="Leo Forsberg" minLength="3" maxLength="40" onChange={handleUsername} value={username} />
          <br />
          <br />
          <Button type="submit" variant="contained" color="primary">LOGIN</Button>
        </form>
      </div>
    </div>
  );
};

export default Login;
