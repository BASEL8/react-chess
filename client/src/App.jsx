/* import React, { useEffect } from 'react';
import axios from 'axios';
import './App.css';
import Lobby from './components/Lobby';

const App = () => {
  useEffect(() => {
    axios.get('http://localhost:8000/api/seeks').then((res) => {
      console.log(res);
    });
  }, []);
  useEffect(() => {
    axios.post('http://localhost:8000/api/seeks/').then((res) => {
      console.log(res);
    });
  }, []);
  useEffect(() => {
    axios.post('http://localhost:8000/api/seeks/333').then((res) => {
      console.log(res);
    });
  }, []);
  useEffect(() => {
    axios.get('http://localhost:8000/api/game/333').then((res) => {
      console.log(res);
    });
  }, []);
  useEffect(() => {
    axios.post('http://localhost:8000/api/game/333/2_3').then((res) => {
      console.log(res);
    });
  }, []);


  return (
    <div className="App">
      <header className="App-header">
        <h1>
          Let's play chess!
        </h1>
      </header>
      <Lobby />
    </div>
  );
}; */

import React from 'react';
import './App.css';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Login from './components/Login';
import Lobby from './components/Lobby';

const App = () => (
  <div className="App">
    <Router basename={process.env.PUBLIC_URL}>
      <Route exact path="/" component={Login} />
      <Route path="/lobby" component={Lobby} />
    </Router>
  </div>
);

export default App;
