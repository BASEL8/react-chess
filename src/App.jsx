import React, { useEffect, useState } from 'react';
import Chess from 'chess.js';
import Chessboard from 'chessboardjsx';
import './App.css';
let chess = new Chess();
let pgn = chess.pgn();

function App() {
  let [fen, updateFen] = useState('start');
  let [moveHistory, updateMoveHistory] = useState([]);
  let newFen = '';

  function updateView() {
    newFen = chess.fen();
    updateFen(newFen);
  }

  function onDrop(sourceSquare) {
    let sourceSq = sourceSquare.sourceSquare;
    let targetSq = sourceSquare.targetSquare;
    let piece = sourceSquare.piece;
    let moveObject = {};
    let newFen = '';

    // console.log(sourceSq, targetSq, piece);

    moveObject = chess.move({from: sourceSq, to: targetSq});

    // console.log(moveObject);

    if(moveObject) {
      console.log('Legal move.');
      updateView();
      let updatedHistory = chess.history();
      updateMoveHistory(updatedHistory);
    }
    else {
      console.log('ILLEGAL MOVE');
    }
  }

  function onClickReset(e) {
    chess.reset();
    updateView();
  }

  return (
    <div className="App">
      <Chessboard
      position={fen}
      onDrop={onDrop}
      />
      <MovesTable moveHistory={moveHistory}/>
      <button onClick={onClickReset}>Restart</button>
    </div>
  );
}

function MovesTable(props) {
  let moveHistory = props.moveHistory;
  console.log(moveHistory);

  let tableRows = moveHistory.map( move => {
    return <tr><td>{move}</td></tr>;
  });

  return(
    <table>
      <thead>
        <tr>
          <th>Moves</th>
        </tr>
      </thead>
      <tbody>
        {tableRows}
      </tbody>
    </table>
  );
}

export default App;
