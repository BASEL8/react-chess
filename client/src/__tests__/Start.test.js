import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../../src/App';
import Login from '../components/Start';
import Lobby from '../components/Lobby';

const onClick = jest.fn();
const mockHandleClick = jest.fn();
const onSearchMock = jest.fn();
const onSubmit = jest.fn();

configure({ adapter: new Adapter() });
render(<App />);

afterEach(cleanup);

describe('Start component', () => {
  it('matches the snapshot', () => {
    const start = create(<Login />);
    expect(start.toJSON()).toMatchSnapshot();
  });

  it('renders input field for user Id', () => {
    const { getByTestId } = render(<Login />);
    const input = getByTestId('userIdInput');
    expect(input).toBeInTheDocument();
  });

  it('should call onChange prop with input value', () => {
    const setup = () => {
      const utils = render(<Login />);
      const input = utils.getByTestId('userIdInput');
      return {
        input,
        ...utils
      }
    }

    const { input } = setup();
    fireEvent.change(input, { target: { value: 'player'} });
    expect(input.value).toBe('player');
  });

  it('renders Enter button', () => {
    const { getByTestId } = render(<Login />);
    const button = getByTestId('button');
    expect(button).toBeInTheDocument();
    expect(button.textContent).toBe('Enter');
  });
});

describe('save to localStorage', () => {
  it('saves user input to localStorage', () => {
    localStorage.setItem = jest.fn();
    localStorage.getItem = jest.fn(() => 'testUser');

    localStorage.setItem('userId','testUser');
    expect(localStorage.getItem('userId')).toBe('testUser');
  });

  it('saves user input to localStorage', () => {
    localStorage.setItem = jest.fn();
    localStorage.getItem = jest.fn(() => '');

    localStorage.setItem('userId','{test: "test"}');
    expect(localStorage.getItem('userId')).toBe('{test: "test"}');
  });
});