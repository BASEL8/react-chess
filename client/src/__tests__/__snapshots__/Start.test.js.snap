// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start component matches the snapshot 1`] = `
<main
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
  />
  <div
    className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
  >
    <div
      className="makeStyles-paper-3"
    >
      <svg
        className="makeStyles-logo-5"
      >
        crown.svg
      </svg>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Chess
      </h1>
      <form
        className="makeStyles-form-6"
        data-testid="form"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="userId"
            id="userId-label"
          >
            Please enter your user ID
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="userId-helper-text"
              aria-invalid={false}
              autoComplete="name"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="userIdInput"
              disabled={false}
              id="userId"
              name="userId"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-229 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-230"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
            id="userId-helper-text"
          >
            User ID must be at least 5 characters long
          </p>
        </div>
        <p
          style={
            Object {
              "color": "red",
            }
          }
        >
          
        </p>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-7 MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Enter
          </span>
        </button>
        <div
          className="MuiBox-root MuiBox-root-289"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            Copyright © 
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="https://github.com/BASEL8/react-chess"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Chess Online
            </a>
            2020
          </p>
        </div>
      </form>
    </div>
  </div>
</main>
`;
